packages/jest-haste-map/src/index.js

```diff
@@ -251,11 +255,14 @@ class HasteMap extends EventEmitter {
           'deprecated. Provide a RegExp instead. See https://github.com/facebook/jest/pull/4063.',
       );
     }
+    const rootDirHash = crypto.createHash('md5').update(options.rootDir);
     this._cachePath = HasteMap.getCacheFilePath(
       this._options.cacheDirectory,
-      `haste-map-${this._options.name}`,
+      `haste-map-${this._options.name}-${rootDirHash}`,
```

Comment:
I removed the rootDir from the roots when generating the hash for the haste map name in the cache, but that can cause conflicts between different projects. Using the hash of the rootDir as a separate fragment of the name prevents this conflicts and allows us to easily reconstruct the cache from a file generated remotely using a different rootDir.